docs/debugging-templates.md

```diff
@@ -0,0 +1,64 @@
+# Debugging Templates
+
+Module templates can be debugged visually through use of a `TemplateDebugView`
+that renders a single module with a series of test cases.
+
+Consider an example module that renders a simple hyperlink. This module
+requires a target URL and also accepts optional link text. A simple Jinja
+template for this module might look like this:
+
+```html
+<a href="{{ value.url }}">
```

Comment:
The location of module templates tends to be unpredictable enough that this feels like it'd be more confusing than helpful to me. We have templates that follow all of these patterns:

- myapp/templates/myapp/module.html - Django template in app directory
- myapp/jinja2/myapp/module.html - Jinja2 template in app directory
- jinja2/v1/_includes/organisms/module.html - Jinja2 template in centralized directory

For that reason I'm inclined to omit this suggestion.

Reply:
That's why I made it match the location you specified in the `wagtail_hooks.py` example below. Seemed like the examples in this doc should be internally consistent.

Reply:
Thanks, I see what you mean.

In practice, though, these would be different, if you're intending the comment to be the actual path to the template file. The full file location of the template might be myapp/jinja2/myapp/link.html, but, because you have to pass the path relative to Django's template search path, you'd actually call refer to it in Python as just template myapp/link.html.

I'd rather avoid putting any template path in the HTML file because it implies endorsement of a certain location, which we don't want to do. My mistake was putting `'path/to/myapp/templates/link.html'` in the example Python code to begin with. I'll change it to simply `'myapp/link.html'`.